/*
*  This file is part of aasdk library project.
*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
*
*  aasdk is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 3 of the License, or
*  (at your option) any later version.

*  aasdk is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with aasdk. If not, see <http://www.gnu.org/licenses/>.
*/

syntax="proto2";

import "TouchConfigData.proto";
import "HapticFeedbackTypeEnum.proto";

package oaa.proto.data;

message InputChannel
{
    repeated uint32 supported_keycodes = 1;
    repeated TouchConfig touch_screen_config = 2;
    repeated TouchConfig touch_pad_config = 3;
    repeated enums.HapticFeedbackType.Enum supported_haptic_feedback = 4; // APK: vxm.java
    optional uint32 max_touchscreen_pointers = 5;  // APK: vya field 5, uint32
}
